compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1243,6 +1252,20 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         }
     }
 
+    fn dectect_wrong_primitive(path: &[Segment]) -> Option<Symbol> {
+        let name = path[path.len() - 1].ident.name;
+        match name {
+            sym::byte => Some(sym::u8),
```

Comment:
Here is a subtle problem: Java has only signed types, but `u8` is unsigned. But I can see the reason for this inconsistency, as a byte really should be unsigned.
So I'm fine with it, but I want to leave the note.